src/lib.rs

```diff
@@ -10,12 +18,40 @@ pub fn verify_plonk_bn254(
     vkey_hash_str: &str,
     committed_values_digest_str: &str,
 ) -> u32 {
-    // TODO: sanity-check the inputs by parsing the build_dir_str, vkey_hash_str, and committed_values_digest_str
+    let proof_str = strip_hex_prefix(proof_str);
+    let vkey_hash_str = strip_hex_prefix(vkey_hash_str);
+    let committed_values_digest_str = strip_hex_prefix(committed_values_digest_str);
+
+    // Decode the hex-encoded string for public values
+    let decoded_bytes = hex::decode(committed_values_digest_str).expect("Invalid committed values field");
+
+    // Check the bit length (bytes * 8 should be 256 bits), hash if necessary
+    let bit_length = decoded_bytes.len() * 8;
+
+    let public_inputs: String = if bit_length > 256 {
```

Comment:
I think it would be better if we always unconditionally did the `Sha256` hashing here, since this heuristic would be confusing to callers.

What about removing the if and the `assets/fibonacci_fixture.json`?

Reply:
I've opened https://github.com/argumentcomputer/lurk-hs/issues/5